test/spec.yml

```diff
@@ -1373,3 +1374,346 @@
     -  in: bn create node8 testArrayOpts --config=aaa --config=bbb
        exit: 1
        err: "Non object create configuration options: aaa,bbb"
+
+- test: Test new trigger (good-path)
+  cleanup:
+    - bn trigger remove testtrigger
+    - bn remove testtriggerfunc
+  steps:
+    - in: bn create node8 testtriggerfunc
+      out: |-
+          Created function * in /home/dockeruser/test
+            (use "bn deploy testtriggerfunc" to deploy the function)
+    - in: bn deploy testtriggerfunc
+      out: |-
+        Deployed function testtriggerfunc
+        Invoke with one of:
+          "bn invoke testtriggerfunc"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: bn trigger upsert testtrigger --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func testtriggerfunc
+      out: CREATED
+
+- test: Test update trigger (good-path)
+  cleanup:
+    - bn trigger remove testupdatetrigger
+    - bn remove testupdatetriggerfunc
+    - bn remove testupdatetriggerfunc2
+  steps:
+    - in: bn create node8 testupdatetriggerfunc
+      out: |-
+          Created function * in /home/dockeruser/test
+            (use "bn deploy testupdatetriggerfunc" to deploy the function)
+    - in: bn deploy testupdatetriggerfunc
+      out: |-
+        Deployed function testupdatetriggerfunc
+        Invoke with one of:
+          "bn invoke testupdatetriggerfunc"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: |-
+        cat <<EOF >/home/dockeruser/test/binaris.yml
+        functions:
+          testupdatetriggerfunc:
+            file: function.js
+            entrypoint: handler
+            executionModel: concurrent
+            runtime: node8
+          testupdatetriggerfunc2:
+            file: function.js
+            entrypoint: handler
+            executionModel: concurrent
+            runtime: node8
+        EOF
+    - in: bn deploy testupdatetriggerfunc2
+      out: |-
+        Deployed function testupdatetriggerfunc2
+        Invoke with one of:
+          "bn invoke testupdatetriggerfunc2"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: bn trigger upsert testupdatetrigger --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func testupdatetriggerfunc
+      out: CREATED
+    - in: bn trigger upsert testupdatetrigger --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func testupdatetriggerfunc2
```

Comment:
Could have just changed the URI to use a different stream name instead of changing the function, would save the time of deploying and removing the second function.

Reply:
Ack